### ISSUE TITLE
Including base-64-encoded image in json body of a .POST triggering crash every time in Xcode 6.4

### ISSUE THREAD

**Issue by genu1** [Alamofire/Alamofire#738]
**Added notes:**

- the image is optional; users generated without image is fine
- breakpoint verified data is not nil 

**Code:**

```swift
var params: [String : AnyObject] = [:]
params["email"] = email
params["password"] = password
params["first_name"] = firstName
params["last_name"] = lastName

// Including a base 64 encoded image is triggering a crash in xcode 6.4
if let image = avatar {
    var imageData = UIImagePNGRepresentation(image).base64EncodedDataWithOptions(
        .Encoding64CharacterLineLength
    )
    params["image"] = imageData
}

Alamofire.request(.POST, request, parameters: params, encoding: .JSON, headers: self.defaultHeaders())
    .response { request, response, data, error in
         print(error)
    }
```

**Comment by cnoon** [comment]
Hi @genu1,

I just added tests around this behavior in 33dcfd4b demonstrating how to do this. I'm not exactly sure why you are crashing, but Alamofire can handle this case without issue. I'm going to close this issue out for now. If you provide more info that leads us to believe there is an issue in Alamofire, we'll be happy to re-open.

Cheers. 🍻

**Comment by andreponce** [comment]
I have the same problem on the simulator. I have no tested on a real device yet.
The problem is on the size of the image when converted to base 64, if the image is too large the string will be too large and json will break. When I reduced the size of the image it worked well.
Working with a 500xH image it broke, but when I reduced the same image to 300xH it worked.
Apparently the problem is in the Alamofire, there's nothing I can do about it.
